redux/skillRatingsSlice.ts

```diff
@@ -6,7 +6,7 @@ export type SkillRatingsState = {
 };
 
 export type SkillRatingsRow = {
-  userSkillId: String;
+  userSkillId: String|number;
```

Comment:
Why is this needed?

Reply:
I generated a random number between 0 and 1 and assigned it as a "placeholder" userSkillId in the transformSkillsAndRatings function.  This is to account for the number type of that function. The userSkillId is the id for the user's particular assignment rating.  When I am displaying the ratings initially in the transformSkillsAndRatings function, if students don't have a rating for a particular skill I need to set that rating to 0.  But in this case those id's have not yet been generated by hasura and need to be created in this function through that "placeholder" userSkillId so that the slider bars work (rating is attached to a particular skill) on the display.  If a user changes these ratings on the page through using the slider bar and I want to save them back to hasura, I cannot save the "placeholder" id.  I have to skip the userSkillId (randomly generated number placeholder) in the transformSkillRatingsForDB function.  I identify whether the userSkillId has been created by hasura or needs to be skipped through checking if it's a number or not in the transformSkillRatingsForDB function.  

This number type accounts for the randomly generated userSkillIds.

Does this make sense?